responsemanager/queryexecutor/queryexecutor.go

```diff
@@ -0,0 +1,300 @@
+package queryexecutor
```

Comment:
+1 for this test -- it's looking good and we haven't had this up till now so I'm excited about that